File: frontend/src/types/schema.ts
export interface SchemaFormData {
  schema_name: string;
  user_name: string;
  schema_element: string;
  ifNotExists: boolean;
}

export const initialSchemaFormData: SchemaFormData = {
  schema_name: '',
  user_name: '',
  schema_element: '',
  ifNotExists: true,
};

File: src/backend/app/models/schema.py
from pydantic import BaseModel
from typing import Optional
from app.models.database import ConnectionInfo


class SchemaListRequest(BaseModel):
    connection: ConnectionInfo
    database_name: str


class SchemaListResponse(BaseModel):
    schemas: list[str]


class CreateSchemaRequest(BaseModel):
    connection: ConnectionInfo
    database_name: str
    schema_name: str
    user_name: Optional[str] = None
    schema_element: Optional[str] = None
    if_not_exists: bool = True


class DeleteSchemaRequest(BaseModel):
    connection: ConnectionInfo
    database_name: str
    name: str
